File: backend/models/PlatformSettings.js
import mongoose from "mongoose";

const PlatformSettingsSchema = new mongoose.Schema({
  maintenanceMode: { type: Boolean, default: false },
  platformName: { type: String, default: "SkillSwap" },
  allowNewRegistrations: { type: Boolean, default: true },
  systemNotice: { type: String, default: "" },
});

export default mongoose.model("PlatformSettings", PlatformSettingsSchema);

File: backend-spring/src/main/java/com/skillswap/backend/model/PlatformSettings.java
package com.skillswap.backend.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Document(collection = "platformsettings")
public class PlatformSettings {
    @Id
    private String id;
    private String platformName = "SkillSwap";
    private boolean maintenanceMode = false;
    private String systemNotice = "";
}
